src/lib/modernExtend.ts

```diff
@@ -100,12 +110,17 @@ export function identify(): ModernExtend {
 
 export interface OnOffArgs {
     powerOnBehavior?: boolean, ota?: DefinitionOta, skipDuplicateTransaction?: boolean, endpoints?: {[s: string]: number},
-    configureReporting?: boolean,
+    configureReporting?: boolean, endpointNames?: string[],
 }
 export function onOff(args?: OnOffArgs): ModernExtend {
     args = {powerOnBehavior: true, skipDuplicateTransaction: false, configureReporting: true, ...args};
 
-    const exposes: Expose[] = args.endpoints ? Object.keys(args.endpoints).map((ep) => e.switch().withEndpoint(ep)) : [e.switch()];
+    let exposes: Expose[];
```

Comment:
Do you already need this, otherwise we can move this to another PR? I propose to first get rid of `endpoints?: {[s: string]: number}` to not have similar arguments (required refactoring all users to use `deviceEndpoint()`)

Reply:
I wanted to use it in my refactoring, but this makes sense too